scripts/base/DeployValidlyBase.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import {Script} from 'forge-std/Script.sol';
+import 'forge-std/StdJson.sol';
+import 'forge-std/console.sol';
+
+import {SovereignPoolFactory} from "@valantis-core/pools/factories/SovereignPoolFactory.sol";
+import {SovereignPool} from "@valantis-core/pools/SovereignPool.sol";
+import {ProtocolFactory} from "@valantis-core/protocol-factory/ProtocolFactory.sol";
+import {Validly} from "src/Validly.sol";
+import {ValidlyFactory} from "src/ValidlyFactory.sol";
+
+abstract contract DeployValidlyBase is Script {
+  using stdJson for string;
+
+  ValidlyFactory public factory;
+  ProtocolFactory protocolFactory;
+  SovereignPoolFactory poolFactory;
+
+  function _deployFactory() public {
+    protocolFactory = new ProtocolFactory(vm.envAddress('SENDER'));
+
+    poolFactory = new SovereignPoolFactory();
+
+    protocolFactory.setSovereignPoolFactory(address(poolFactory));
```

Comment:
protocol factory and sovereign pool factory only need to be deployed once per chain. Consider this when deploying in production.